### PR TITLE
Add the Reverse function

### PR DESCRIPTION
**Pull request by shonk-msft**
Add the `Reverse` function and improve "pinging" the exec ctx.

### CODE REVIEW

File: src/Core/Rexl.Code/Operations/Funcs/Sequence.cs [hunk 1]
```diff
@@ -2,10 +2,13 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net.NetworkInformation;
```
Comment (shonk-msft) [inline src/Core/Rexl.Code/Operations/Funcs/Sequence.cs:8]: >NetworkInformation [](http://example.com/codeflow?start=17&length=18)

Oops.

File: src/Core/Rexl.Bind/Operations/Funcs/Sequence.cs [hunk 2]
```diff
@@ -245,6 +245,94 @@ protected override bool IsUnboundedCore(BndCallNode call)
     }
 }
 
+/// <summary>
+/// Sort sequence function.
```
Comment (shonk-msft) [inline src/Core/Rexl.Bind/Operations/Funcs/Sequence.cs:249]: >Sort [](http://example.com/codeflow?start=4&length=4)

Oops.

File: src/Core/Rexl.Code/Operations/Funcs/Sequence.cs [hunk 3]
```diff
@@ -14,11 +17,362 @@
...
+
+    private static IEnumerable<T> Exec<T>(IEnumerable<T> src, ExecCtx ctx, int id)
+    {
+        Validation.AssertValueOrNull(src);
+        Validation.AssertValue(ctx);
+
+        if (src is null)
+            return null;
+
+        // REVIEW: Optimize for various cases. For example, ICollection, ICursorable.
```
Comment (shonk-msft) [inline src/Core/Rexl.Code/Operations/Funcs/Sequence.cs:61]: >REVIEW [](http://example.com/codeflow?start=11&length=6)

Remove.

File: src/Core/Rexl.Code/Operations/MultiFormOperGen.Exec.cs [hunk 4]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
```
Comment (shonk-msft) [inline src/Core/Rexl.Code/Operations/MultiFormOperGen.Exec.cs:7]: >ObjectModel [](http://example.com/codeflow?start=25&length=11)

Oops.

File: src/Core/Rexl.Code/Operations/Funcs/Sequence.cs [hunk 5]
```diff
@@ -19,6 +21,372 @@ namespace Microsoft.Rexl.Code;
...
+        {
+            if (!_counter.TryGetCount(out long count))
+                count = _counter.GetCount(_ping);
+
+            var curs = _src.GetCursor();
+            for (long i = count; --i >= 0;)
+            {
+                bool tmp = curs.MoveTo(i, _ping);
+                Validation.Assert(tmp);
+                yield return curs.Current;
```
Comment (shonk-msft) [inline src/Core/Rexl.Code/Operations/Funcs/Sequence.cs:203]: >Current [](http://example.com/codeflow?start=34&length=7)

Slightly better to use `Value`.

File: src/Core/Rexl.Code/Operations/Funcs/Sequence.cs [hunk 6]
```diff
@@ -19,6 +21,372 @@ namespace Microsoft.Rexl.Code;
...
+            : base()
+        {
+            Validation.AssertValue(items);
+
+            _items = items;
+        }
+
+        public override IEnumerator<T> GetEnumerator()
+        {
+            int count = _items.Count;
```
Comment (shonk-msft) [inline src/Core/Rexl.Code/Operations/Funcs/Sequence.cs:312]: >Count [](http://example.com/codeflow?start=31&length=5)

A bit better to call this once up front.

**Review by askwenhan — APPROVED** [review]

**Pull request merged**